nexus/socialMedia.ts

```diff
@@ -0,0 +1,82 @@
+import { schema } from 'nexus'
+import { SocialMedia } from '@prisma/client'
+import { NotAuthorizedError, UserInputError } from './errors'
+import { Context } from './utils'
+import { getSocialMediaUpdate, prepareNewSocialMedia } from './utils/socialMedia'
+
+schema.objectType({
+  name: 'SocialMedia',
+  definition(t) {
+    t.model.id()
+    t.model.platform()
+    t.model.url()
+  },
+})
+
+schema.extendType({
+  type: 'Query',
+  definition(t) {
+    t.list.field('socialMedia', {
+      type: 'SocialMedia',
+      resolve: async (_parent, args, ctx: Context): Promise<SocialMedia[]> => {
+        return ctx.db.socialMedia.findMany({
+          where: {
+            authorId: _parent.id,
+          },
+        })
+      },
+    })
+  },
+})
+
+schema.extendType({
+  type: 'Mutation',
+  definition(t) {
+    t.list.field('updateSocialMedia', {
+      type: 'SocialMedia',
+      args: {
+        facebook: schema.stringArg({ required: false }),
+        instagram: schema.stringArg({ required: false }),
+        youtube: schema.stringArg({ required: false }),
+        website: schema.stringArg({ required: false }),
+        linkedin: schema.stringArg({ required: false }),
+      },
+      resolve: async (_parent, args, ctx: Context): Promise<SocialMedia[]> => {
```

Comment:
We can safely rely on the TypeScript compiler for the `ctx` and return types and we prefer to only add types where necessary just so that if we refactor things it's one less thing to change